lib/tasks/play.js

```diff
@@ -34,14 +48,24 @@ class TaskPlay extends Task {
             await this.playToConfMember(this.ep, memberId, confName, confUuid, this.url);
           }
         } else {
-          const file = (this.timeoutSecs >= 0 || this.seekOffset >= 0) ?
-            {file: this.url, seekOffset: this.seekOffset, timeoutSecs: this.timeoutSecs} : this.url;
+          let file = this.url;
+          if (this.seekOffset >= 0) {
+            file = {file: this.url, seekOffset: this.seekOffset};
+            this.seekOffset = -1;
+          }
           const result = await ep.play(file);
-          await this.performAction(Object.assign(result, {reason: 'playCompleted'}),
-            !(this.parentTask || cs.isConfirmCallSession));
+          playbackSeconds += parseInt(result.playbackSeconds);
+          playbackMilliseconds += parseInt(result.playbackMilliseconds);
+          if (cs.callStatus === CallStatus.Completed || !this.loop || completed) {
```

Comment:
the proper way to check if the caller hung (or at least to be consistent with other code) is simply to check if the task has been killed, ie `this.killed`.  Because when the caller hangs up we do kill the currently-executing task.  So I would like to see that change (though you can still expose the new CallStatus getter on the CallSession class)

Reply:
Change made, sorry I should've spotted that. I was thinking the completed flag might have become unnecessary as the timeout calls `this.kill`, but I still thinks it required for the case where timeout or loop haven't been set.